packages/starknet-snap/src/getErc20TokenBalance.ts

```diff
@@ -50,13 +53,21 @@ export async function getErc20TokenBalance(params: ApiParams) {
       `getErc20Balance:\nerc20Address: ${erc20Address}\nuserAddress: ${userAddress}`,
     );
 
-    const resp = await callContract(network, erc20Address, 'balanceOf', [
-      numUtils.toBigInt(userAddress).toString(10),
-    ]);
+    // Use LATEST block for non-deployed accounts to avoid showing non-zero balances
+    // from pending transactions, as deployment requires a confirmed non-zero balance.
```

Comment:
the comment seem not reference to the use case?
(dont copy, but just the meaning)

case 1: when account is not deployed, we use latest block to make sure the balance is correct/stable, which is the same behaviour when discovering the account is deploy required
case 2: when account is deployed, we use pending block to make sure we can show the balance update asap to unblock concurrent transaction